### PR TITLE
Add mish op to ngraph

### PR DESCRIPTION
**Pull request by iimironov**

### CODE REVIEW

File: ngraph/src/ngraph/op/mish.cpp [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    }
+
+    bool evaluate_mish(const HostTensorPtr& arg0, const HostTensorPtr& out, const size_t count)
+    {
+        bool rc = true;
+        out->set_unary(arg0);
+
+        switch (arg0->get_element_type())
+        {
+            TYPE_CASE(boolean)(arg0, out, count);
```
Comment (ilyachur) [inline ngraph/src/ngraph/op/mish.cpp:68]: Do we need boolean data type for this operation?

Please check that evaluate method contains only data types which are needed.

File: ngraph/src/ngraph/op/mish.hpp [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+#pragma once
+
+#include "ngraph/node.hpp"
+#include "ngraph/op/op.hpp"
+
+namespace ngraph
+{
+    namespace op
+    {
+        namespace v1
```
Comment (lazarevevgeny) [inline ngraph/src/ngraph/op/mish.hpp:26]: namespace should be "v4"

File: ngraph/src/ngraph/op/mish.hpp [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        {
+            /// \brief A Self Regularized Non-Monotonic Neural Activation Function
+            /// x <  0 => f(x) =  x * tanh(log(exp(x) + 1.))
```
Comment (lazarevevgeny) [inline ngraph/src/ngraph/op/mish.hpp:29]: redundant space character

File: ngraph/src/ngraph/runtime/reference/mish.hpp [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    namespace runtime
+    {
+        namespace reference
+        {
+            template <typename T>
+            void mish(const T* arg, T* out, size_t count)
+            {
+                for (size_t i = 0; i < count; i++)
+                {
+                    out[i] = arg[i] < 0 ? arg[i] * std::tanh(std::log((std::exp(arg[i]) + 1.0)))
```
Comment (lazarevevgeny) [inline ngraph/src/ngraph/runtime/reference/mish.hpp:33]: The operation implementation contradicts with the specification!

In the specification there is no condition on the `arg[0]`. Either the specification or implementation is wrong.

File: ngraph/src/ngraph/op/mish.hpp [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+            /// x >= 0 => f(x) = x
```
Comment (lazarevevgeny) [inline ngraph/src/ngraph/op/mish.hpp:30]: The comment does not correspond to the specification.

File: ngraph/test/op_eval/mish.cpp [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+using namespace ngraph;
+
+TEST(op_eval, mish_0D)
+{
+    auto p = make_shared<op::Parameter>(element::f32, Shape{});
+    auto mish = make_shared<op::v1::Mish>(p, element::f32);
+    auto fun = make_shared<Function>(OutputVector{mish}, ParameterVector{p});
+
+    std::vector<std::vector<float>> inputs{{-1.0}, {1.0}, {20.0}};
+    std::vector<std::vector<float>> expected_result{{0}, {0}, {0}};
```
Comment (lazarevevgeny) [inline ngraph/test/op_eval/mish.cpp:39]: Can we have a test input where the expected result is no 0?

File: ngraph/test/type_prop/mish.cpp [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+TEST(type_prop, mish)
+{
+    auto data = make_shared<op::Parameter>(element::f32, Shape{1, 3, 6});
+    auto mish_func = make_shared<op::v1::Mish>(data);
+    EXPECT_EQ(mish_func->get_element_type(), element::f32);
+    EXPECT_EQ(mish_func->get_shape(), (Shape{1, 3, 6}));
+}
+
+TEST(type_prop, mish_partial)
```
Comment (lazarevevgeny) [inline ngraph/test/type_prop/mish.cpp:32]: Lets add a test where the input rank is static and just one of the dimensions is undefined.

File: ngraph/test/op_eval/mish.cpp [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+    for (size_t i = 0; i < inputs.size(); i++)
+    {
+        auto result = make_shared<HostTensor>();
+        ASSERT_TRUE(
+            fun->evaluate({result}, {make_host_tensor<element::Type_t::f32>(Shape{}, inputs[i])}));
+        EXPECT_EQ(result->get_element_type(), element::f32);
+        EXPECT_EQ(result->get_shape(), (Shape{}));
+        auto result_data = read_vector<float>(result);
+        EXPECT_NEAR(result_data[0], expected_result[i][0], 0.3);
```
Comment (lazarevevgeny) [inline ngraph/test/op_eval/mish.cpp:49]: Why the precision is 0.3? It is so big value! It should be ~1e-6!

File: ngraph/src/ngraph/opsets/opset4_tbl.hpp [hunk 9]
```diff
@@ -153,4 +154,4 @@ NGRAPH_OP(TopK, ngraph::op::v3)
 
 // New operations added in opset4
 NGRAPH_OP(NonMaxSuppression, ngraph::op::v4)
-NGRAPH_OP(Acosh, ngraph::op::v3)
```
Comment (lazarevevgeny) [inline ngraph/src/ngraph/opsets/opset4_tbl.hpp:156]: Why did you remove this line?

Comment (ilyachur) [reply ngraph/src/ngraph/opsets/opset4_tbl.hpp:156]: This line was moved to new place (in alphabet order).
